dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESMappingAPIImpl.java

```diff
@@ -464,203 +546,216 @@ protected void loadPermissions(final Contentlet con, final Map<String,Object> m)
 					}
 				}
 
 				// Store sha256 hash for unique fields in the index
-				if (f.isUnique() && m.containsKey(keyName)) {
-					final Object uniqueValue = m.get(keyName);
-					m.put(keyName + ESUtils.SHA_256,
-							ESUtils.sha256(keyName, uniqueValue, con.getLanguageId()));
+				if (field.isUnique() && contentletMap.containsKey(keyName)) {
+					final Object uniqueValue = contentletMap.get(keyName);
+					contentletMap.put(keyName + ESUtils.SHA_256,
+							ESUtils.sha256(keyName, uniqueValue, contentlet.getLanguageId()));
 				}
-
 			} catch (Exception e) {
-				Logger.warn(ESMappingAPIImpl.class, "Error indexing field: " + f.getFieldName()
-						+ " of contentlet: " + con.getInode(), e);
+				Logger.warn(ESMappingAPIImpl.class, "Error indexing field: " + field.getFieldName()
+						+ " of contentlet: " + contentlet.getInode(), e);
 				throw new DotDataException(e.getMessage(),e);
 			}
 		}
 	}
 
+	private boolean loadTagsField(final Contentlet contentlet,
+								  final Map<String, Object> contentletMap,
+								  final Structure structure,
+								  final String keyName) throws DotDataException {
+
+		final List<Tag> tagList = APILocator.getTagAPI().getTagsByInode(contentlet.getInode());
+		if(tagList ==null || tagList.size()==0) {
+			return true;
+		}
+
+		final List<String> tagsNames = tagList.stream().map(Tag::getTagName).collect(
+				Collectors.toList());
+
+		contentletMap.put(keyName, tagsNames);
+		contentletMap.put(ESMappingConstants.TAGS, tagsNames);
+
+		if ( Structure.STRUCTURE_TYPE_PERSONA != contentlet.getStructure().getStructureType() ) {
+			final List<String> personaTagsNames = tagList.stream()
+					.filter(Tag::isPersona)
+					.map(Tag::getTagName)
+					.collect(Collectors.toList());
+
+			contentletMap.put(structure.getVelocityVarName() + "."
+					+ ESMappingConstants.PERSONAS, personaTagsNames);
+			contentletMap.put(ESMappingConstants.PERSONAS, personaTagsNames);
+		}
+
+		return false;
+	}
+
+	private void loadKeyValueField(final Map<String, Object> contentletMap,
+								   final String keyName,
+								   final TikaUtils tikaUtils,
+								   final Field field,
+								   final String valueObj,
+								   final boolean fileMetadata) throws DotDataException, DotSecurityException {
+
+		final Map<String,Object> keyValueMap = KeyValueFieldUtil.JSONValueToHashMap(valueObj);
+		Set<String> allowedFields = new HashSet<>();
+
+		if(fileMetadata) {
+			// http://jira.dotmarketing.net/browse/DOTCMS-7243
+			final List<FieldVariable> fieldVariables = APILocator.getFieldAPI().getFieldVariablesForField(
+					field.getInode(), APILocator.getUserAPI().getSystemUser(), false);
+			for(final FieldVariable fieldVariable : fieldVariables) {
+				if(fieldVariable.getKey().equals(ESMappingConstants.DOT_INDEX_PATTERN)) {
+
+					final String[] names = fieldVariable.getValue().split(",");
+					allowedFields        = new HashSet<>();
+					for(final String name : names)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [This statement should have braces](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4960665)